src/modules/fixEmbedTwitterVideo/fixEmbedTwitterVideo.module.ts

```diff
@@ -0,0 +1,173 @@
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, MessageType } from 'discord.js';
+import { z } from 'zod';
+
+import { createModule } from '../../core/createModule';
+import { resolveCatch } from '../../helpers/resolveCatch.helper';
+
+const FXTwitterResponseSchema = z.object({
+  code: z.number(),
+  message: z.string(),
+  tweet: z.object({
+    media: z
+      .object({
+        videos: z
+          .array(
+            z
+              .object({
+                type: z.enum(['gif', 'video']),
+                url: z.string(),
+              })
+              .optional(),
+          )
+          .optional(),
+      })
+      .optional(),
+  }),
+});
+
+type URLMapping = {
+  pattern: RegExp;
+  replacement: string;
+};
+
+const modulePrefixButtonId = 'fixEmbedTwitterVideo-';
+const deleteBotAnswerPrefixButtonId = modulePrefixButtonId + 'deleteBotAnswer-';
+const deleteBotAnswerButtonId = deleteBotAnswerPrefixButtonId + '$authorIdMessage';
+const ignoreConfirmationButtonId = modulePrefixButtonId + 'ignoreBotButtons';
+
+const twitterUrlMappings: URLMapping[] = [
+  {
+    pattern: /https?:\/\/(mobile\.)?(twitter|x)\.com\/(\S+)\/status\/(\d+)/g,
+    replacement: 'https://fxtwitter.com/$3/status/$4',
+  },
+];
+
+const FXTwitterUrlMappings: URLMapping[] = [
+  {
+    pattern: /https?:\/\/fxtwitter\.com\/(\S+)\/status\/(\d+)/g,
+    replacement: 'https://api.fxtwitter.com/$1/status/$2',
+  },
+];
+
+const matchAndReplaceTweetLink = (message: string, urlMappings: URLMapping[]) => {
+  let tweetLink = '';
+
+  for (const urlMapping of urlMappings) {
+    const twitterLinks = message.match(urlMapping.pattern);
+
+    if (twitterLinks && twitterLinks.length > 0) {
+      tweetLink = twitterLinks[0].replace(urlMapping.pattern, urlMapping.replacement);
+
+      break;
+    }
+  }
+
+  return tweetLink;
+};
```

Comment:
you can use reduce here

Reply:
u prefer this ? 

```
const matchAndReplaceTweetLink = (message: string, urlMappings: URLMapping[]) => {
  return urlMappings.reduce((acc, urlMapping) => {
    const twitterLinks = message.match(urlMapping.pattern);

    if (twitterLinks && twitterLinks.length > 0) {
      return twitterLinks[0].replace(urlMapping.pattern, urlMapping.replacement);
    }

    return acc;
  }, '');
};
```

Reply:
after reflexion, i don't agree to use a reduce, it will not be clear like the for loop. With my loop, we understand that at the first pattern that match I stop the loop.